core/src/main/java/nl/nn/adapterframework/processors/TransactionAttributePipeProcessor.java

```diff
@@ -41,51 +43,56 @@ public class TransactionAttributePipeProcessor extends PipeProcessorBase {
 	private @Getter @Setter PlatformTransactionManager txManager;
 
 	@Override
-	protected PipeRunResult processPipe(PipeLine pipeLine, IPipe pipe, Message message, PipeLineSession pipeLineSession, ThrowingFunction<Message, PipeRunResult,PipeRunException> chain) throws PipeRunException {
-		PipeRunResult pipeRunResult;
+	protected PipeRunResult processPipe(PipeLine pipeline, IPipe pipe, Message message, PipeLineSession pipeLineSession, ThrowingFunction<Message, PipeRunResult,PipeRunException> chain) throws PipeRunException {
 		TransactionDefinition txDef;
-		int txTimeout=0;
-		if (pipe instanceof HasTransactionAttribute) {
+		int txTimeout = 0;
+		if(pipe instanceof HasTransactionAttribute) {
 			HasTransactionAttribute taPipe = (HasTransactionAttribute) pipe;
 			txDef = taPipe.getTxDef();
-			txTimeout= taPipe.getTransactionTimeout();
+			txTimeout = taPipe.getTransactionTimeout();
 		} else {
-			txDef = SpringTxManagerProxy.getTransactionDefinition(TransactionDefinition.PROPAGATION_SUPPORTS,txTimeout);
+			txDef = SpringTxManagerProxy.getTransactionDefinition(TransactionDefinition.PROPAGATION_SUPPORTS, txTimeout);
 		}
 		IbisTransaction itx = new IbisTransaction(txManager, txDef, "pipe [" + pipe.getName() + "]");
+		boolean isJdbcFacade = pipe instanceof MessageSendingPipe && ((MessageSendingPipe) pipe).getSender() instanceof JdbcFacade;
```

Comment:
Hoe zit het met combinatie van JMS & JDBC? Als er eerder een fout is geweest in een JDBC pipe, geeft daarna een transactionele F&F JMS sender altijd een exception, voor alle queue implementaties die we moeten ondersteuen? Of moeten we daarvoor eenzelfde check doen als voor JDBC Senders?

Ik zou dat graag in een IAF-Test scenario getest zien (ook met WebSphere, voor 7.9, want daar hebben we WebSphereMQ ipv ActiveMQ / ActiveMQ Artemis).

Reply:
We moeten het houden zoals het was, dus bij TX capable senders moet hij altijd de transaction markeren als RollbackOnly.